companyweb_base/models/companyweb_credential_wizard.py

```diff
@@ -0,0 +1,21 @@
+# Copyright 2021-2023 ACSONE SA/NV
+from odoo import fields, models
+from odoo.exceptions import ValidationError
+
+
+class CompanyWebCredentialWizard(models.TransientModel):
+
+    _name = "companyweb_base.credential_wizard"
+    _description = "Ask for CompanyWeb login & password"
+    cweb_login = fields.Char("Company Web Login")
+    cweb_password = fields.Char("Company Web Password")
+
+    def get_credentials(self):
+        self.ensure_one()
+        if not (self.cweb_login or self.cweb_password):
+            raise ValidationError("No input given")
+        self.env.user.cweb_login = self.cweb_login
+        self.env.user.cweb_password = self.cweb_password
+        self.env["res.partner"].browse(
+            self.env.context["active_id"]
+        ).button_companyweb()
```

Comment:
A return here might help with your wizard issues ? Not sure.

Reply:
i tried this solution and many variant this afternoon but without any good result
https://www.odoo.com/forum/help-1/how-to-reload-page-in-python-130677
I also tried to change the way i call the wizard from button_companyweb -> no better result
(change the target, use nodestroy option, end method with a return status, end method with a exception to add a dialogbox, use ir.actions.client with many tags)
I can't have open the wizard more than 1 times
With the debug i can see that the context/self is ok (vat is filled/the wizard_id is known etc)
So for now if the user has no login/password -> wizard
If he enters the right credentials -> ok
if not -> KO and he has to click on more time on the button
Same for the case where the credentials are not recognised -> the wizard is shown 
If he enters the right credentials -> ok
if not -> KO and he has to click on more time on the button
any help is welcome

Reply:
finally a return was the solution